Mathlib/CategoryTheory/Galois/Topology.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Endomorphism
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.Topology.Algebra.Group.Basic
+
+/-!
+
+# Topology of fundamental group
+
+In this file we define a natural topology on the automorphism group of a functor
+`F : C ⥤ FintypeCat`: It is defined as the subspace topology induced by the natural
+embedding of `Aut F` into `∀ X, Aut (F.obj X)` where
+`Aut (F.obj X)` carries the discrete topology.
+
+## References
+
+- Stacks Project: Tag 0BMQ
+
+-/
+universe u₁ u₂ v₁ v₂ v w
+
+namespace CategoryTheory
+
+namespace PreGaloisCategory
+
+open Functor
+
+variable {C : Type u₁} [Category.{u₂} C] (F : C ⥤ FintypeCat.{w})
+
+/-- For a functor `F : C ⥤ FintypeCat`, the canonical embedding of `Aut F` into
+the product over `Aut (F.obj X)` for all objects `X`. -/
+def autEmbedding : Aut F →* ∀ X, Aut (F.obj X) :=
+  MonoidHom.mk' (fun σ X ↦ σ.app X) (fun _ _ ↦ rfl)
+
+@[simp]
+lemma autEmbedding_apply (σ : Aut F) (X : C) : autEmbedding F σ X = σ.app X :=
+  rfl
+
+lemma autEmbedding_injective : Function.Injective (autEmbedding F) := by
+  intro σ τ h
+  ext X x
+  have : σ.app X = τ.app X := congr_fun h X
+  rw [← Iso.app_hom, ← Iso.app_hom, this]
+
+instance (X : C) : TopologicalSpace (F.obj X) := ⊥
+instance (X : C) : DiscreteTopology (F.obj X) := ⟨rfl⟩
+instance (X : C) : TopologicalSpace (Aut (F.obj X)) := ⊥
+instance (X : C) : DiscreteTopology (Aut (F.obj X)) := ⟨rfl⟩
```

Comment:
Should not these be local/scoped instances?

Reply:
Good idea, changed it. I left the other instances as is, since they are rather specific to the situation. The topology on `Aut F` should be the natural one, but please say so if I should scope these as well.